android/java/org/chromium/chrome/browser/rewards/BraveRewardsTipConfirmationFragment.java

```diff
@@ -0,0 +1,183 @@
+            mMonthlyContribution = getArguments().getBoolean(
+                    BraveRewardsSiteBannerActivity.IS_MONTHLY_CONTRIBUTION);
+            int status = getArguments().getInt(BraveRewardsSiteBannerActivity.STATUS_ARGS, -1);
+            String publisherName =
+                    getArguments().getString(BraveRewardsSiteBannerActivity.NAME_ARGS);
+
+            initView(view, status, publisherName, amount, mMonthlyContribution);
+        }
+        return view;
+    }
+
+    @SuppressLint("SetTextI18n")
+    private void initView(
+            View view, int status, String name, double amount, boolean monthlyContribution) {
+        mStatus = status;
+        mPublisherName = name;
+        TextView tipAmountTextView = view.findViewById(R.id.amount_sent);
+        tipAmountTextView.setText(amount + " " + BraveRewardsHelper.BAT_TEXT);
+        TextView tipTypeDescriptionTextView = view.findViewById(R.id.your_tip_amount_sent);
+        TextView oneTtimeTipNote = view.findViewById(R.id.one_time_tip_note);
+        String oneTimeNoteString = getResources().getString(R.string.one_time_tip_success_note);
+
+        final StringBuilder sb1 = new StringBuilder();
+        sb1.append("<b>");
+        sb1.append(getResources().getString(R.string.note_text));
+        sb1.append("</b>");
+        sb1.append(" ");
+        sb1.append(oneTimeNoteString);
+        Spanned toInsert = BraveRewardsHelper.spannedFromHtmlString(sb1.toString());
+        oneTtimeTipNote.setText(toInsert);
+
+        mNextTipDateText = view.findViewById(R.id.next_tip_date_text);
+        mNextTipDateValue = view.findViewById(R.id.next_tip_date_value);
+
+        if (monthlyContribution) {
+            tipTypeDescriptionTextView.setText(
+                    getResources().getString(R.string.your_monthly_tip_has_been_sent));
+        } else {
+            tipTypeDescriptionTextView.setText(
+                    getResources().getString(R.string.your_one_time_tip_has_been_sent));
+            oneTtimeTipNote.setVisibility(View.VISIBLE);
+        }
+        mBraveRewardsNativeWorker.GetReconcileStamp();
+        setTweetClickListener(view);
+    }
+
+    private void setTweetClickListener(View view) {
+        TextView twitter_button = view.findViewById(R.id.twitter_button);
+        twitter_button.setOnClickListener(v -> {
+            String tweetUrl;
+            if (mStatus == BraveRewardsSiteBannerActivity.TIP_SUCCESS) {
+                tweetUrl = getTipSuccessTweetUrl(mPublisherName);
+                Uri uri = Uri.parse(tweetUrl);
+                startActivity(new Intent(Intent.ACTION_VIEW, uri));
+            } else if (mStatus == BraveRewardsSiteBannerActivity.TIP_PENDING) {
+                tweetUrl = getTipPendingTweetUrl(mPublisherName);
+                Uri uri = Uri.parse(tweetUrl);
+                startActivity(new Intent(Intent.ACTION_VIEW, uri));
+            } else {
+                // error
```

Comment:
Same here, worth adding assert and log error

Reply:
added error log